src/odemis/gui/model/__init__.py

```diff
@@ -1554,31 +1568,86 @@ def moveStageToView(self):
-        for ax, p in move.items():
-            axis_def = self._stage.axes[ax]
-            if (hasattr(axis_def, "range") and
-                not axis_def.range[0] <= p <= axis_def.range[1]
-               ):
-                p = max(axis_def.range[0], min(p, axis_def.range[1]))
-                move[ax] = p
-                logging.info("Restricting stage axis %s move to %g mm due to stage limit",
-                             ax, p * 1e3)
-
-        logging.debug("Requesting stage to move to %s m", move)
+        logging.debug("Requesting stage to move to %s mm in x direction and %s mm in y direction",
+                      move["x"] * 1e3, move["y"] * 1e3)
         f = self._stage.moveAbs(move)
         self._fstage_move = f
         f.add_done_callback(self._on_stage_move_done)
         return f
 
+    def clipToStageLimits(self, pos):
+        """
+        Clip current a position in x/y direction to the maximum allowed stage limits.
+
+        :param pos (dict): Position to be clipped with keys "x" and "y"
+        :return(dict): Position clipped to the stage limits with keys "x" and "y"
+        """
+        stage_limits = self._getStageLimitsXY()
+        if not stage_limits["x"][0] <= pos["x"] <= stage_limits["x"][1]:
+            pos["x"] = max(stage_limits["x"][0], min(pos["x"], stage_limits["x"][1]))
+            logging.info("Movements of the stage in x limited to %s mm, restricting movement to %s mm.",
+                         stage_limits["x"], pos["x"] * 1e3)
+
+        if not stage_limits["y"][0] <= pos["x"] <= stage_limits["y"][1]:
+            pos["y"] = max(stage_limits["y"][0], min(pos["y"], stage_limits["y"][1]))
+            logging.info("Movements of the stage in y limited to %s mm, restricting movement to %s mm.",
+                         stage_limits["y"], pos["y"] * 1e3)
+        return pos
+
+    def _getStageLimitsXY(self):
+        """
+        Based on the physical stage limit and the area of the image used for imaging the stage limits are returned in
+        a dict. (MD_POS_ACTIVE_RANGE defines the area which can be used for imaging)
+        If no stage limits are defined an empty dict is returned.
+
+        :return (dictionary): dict which contains the stage limits in x and y direction
+        """
+        stage_limits = {}
+        # Physical stage limits
+        if hasattr(self._stage.axes["x"], "range"):
+            stage_limits["x"] = list(self._stage.axes["x"].range)
+        if hasattr(self._stage.axes["y"], "range"):
+            stage_limits["y"] = list(self._stage.axes["x"].range)
+
+        # Area which can be used for imaging
+        pos_active_range = self._stage.getMetadata().get(model.MD_POS_ACTIVE_RANGE, {})
+        if "x" in pos_active_range:
+            stage_limits = self._updateStageLimits(stage_limits, {"x": pos_active_range["x"]})
+        if "y" in pos_active_range:
+            stage_limits = self._updateStageLimits(stage_limits, {"y": pos_active_range["y"]})
```

Comment:
`pos_active_range` is the right format already. So you can just write:
`stage_limits = self._updateStageLimits(stage_limits, pos_active_range)`

Reply:
I actually only want to update the stage limits in _x/y_ if somehow a limit in _z_ is included in MD_POS_ACTIVE_RANGE it should not update the stage limits with that. This implementation should limit the area in which you can scroll through the image not the focus for example.
By passing the entire dict "_pos_active_range_" that might happen. (It could include more than just those two axis isn't it?)

If we don't care about updating also other axis the implementation should become: 
```
        if "x" in pos_active_range or "y" in pos_active_range:
            stage_limits = self._updateStageLimits(stage_limits, pos_active_range)
```
To prevent updating the same limits twice.